velox/type/parser/tests/TypeParserTest.cpp

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/types/HyperLogLogType.h"
+#include "velox/functions/prestosql/types/JsonType.h"
```

Comment:
It doesn't seem right to have a dependency on prestosql here. CC: @pedroerp

Reply:
@mbasmanova this is a test file. Can you share why it would be not right to depend on prestosql?

Reply:
I see that even a dependency across modules for tests is not ideal. If we decide to keep the implementation here, I can define these types here again.

Reply:
+1. Seems like what we really want to test is whether the parser understands user defined types. Could we rather define a super tiny and simple type in this file, register, then test that we can parse it? Ideally, one without and one with blank spaces in the name